SPDX copyright text

ret2libc opened the pull request:
Partially addresses https://github.com/rizinorg/rizin/issues/683

ret2libc committed:
librz/main: rename from rz_ to rz- for consistency

wargio commented on librz/asm/arch/ppc/libvle/vle.c:
```diff
@@ -1,4 +1,5 @@
-/* rizin - LGPL - Copyright 2017 - wargio */
+// SPDX-FileCopyrightText: 2017 wargio <wargio@libero.it>
```
```suggestion
// SPDX-FileCopyrightText: 2017 deroad <wargio@libero.it>
```

wargio commented on librz/asm/arch/ppc/libps/libps_internal.h:
```diff
@@ -1,3 +1,4 @@
+// SPDX-FileCopyrightText: 2017 wargio <wargio@libero.it>
```
```suggestion
// SPDX-FileCopyrightText: 2017 deroad <wargio@libero.it>
```

wargio commented on librz/asm/arch/ppc/libps/libps.h:
```diff
@@ -1,3 +1,4 @@
+// SPDX-FileCopyrightText: 2017 wargio <wargio@libero.it>
```
```suggestion
// SPDX-FileCopyrightText: 2017 deroad <wargio@libero.it>
```

XVilka commented on librz/analysis/p/analysis_rsp.c:
```diff
@@ -1,3 +1,4 @@
+// SPDX-FileCopyrightText: 2016-2017 bobby.smiles32@gmail.com <bobby.smiles32@gmail.com>
```
Maybe cut to just `bobby.smiles32` as the name?

wargio commented on librz/asm/arch/ppc/libps/libps.c:
```diff
@@ -1,3 +1,4 @@
+// SPDX-FileCopyrightText: 2017 wargio <wargio@libero.it>
```
```suggestion
// SPDX-FileCopyrightText: 2017 deroad <wargio@libero.it>
```

wargio commented on librz/asm/p/asm_x86_vm.c:
```diff
@@ -1,4 +1,5 @@
-/* rizin - LGPL - Copyright 2018 - wargio */
+// SPDX-FileCopyrightText: 2018 wargio <wargio@libero.it>
```
```suggestion
// SPDX-FileCopyrightText: 2018 deroad <wargio@libero.it>
```

XVilka commented on librz/asm/arch/8051/8051_ass.c:
```diff
@@ -1,3 +1,4 @@
+// SPDX-FileCopyrightText: 2009-2019 hmht <55194419+hmht@users.noreply.github.com>
```
Maybe remove all `users.noreply.github.com>` email occurencies? They don't make sense in copyright

wargio replied:
agreed

wargio commented on librz/parse/p/parse_avr_pseudo.c:
```diff
@@ -1,3 +1,4 @@
+// SPDX-FileCopyrightText: 2017-2019 wargio <wargio@libero.it>
```
```suggestion
// SPDX-FileCopyrightText: 2017-2019 deroad <wargio@libero.it>
```

wargio commented on librz/parse/p/parse_sh_pseudo.c:
```diff
@@ -1,3 +1,4 @@
+// SPDX-FileCopyrightText: 2017 wargio <wargio@libero.it>
```
```suggestion
// SPDX-FileCopyrightText: 2017 deroad <wargio@libero.it>
```

XVilka commented on librz/asm/arch/pyc/opcode.h:
```diff
@@ -1,3 +1,5 @@
+// SPDX-FileCopyrightText: 2016-2020 c0riolis <c0riolis>
```
Remove invalid emails like `<name>` please, just name

wargio commented on librz/util/asn1.c:
```diff
@@ -1,4 +1,5 @@
-/* rizin - LGPL - Copyright 2017-2018 - wargio */
+// SPDX-FileCopyrightText: 2017-2018 wargio <wargio@libero.it>
```
```suggestion
// SPDX-FileCopyrightText: 2017-2018 deroad <wargio@libero.it>
```

wargio commented on librz/util/astr.c:
```diff
@@ -1,4 +1,6 @@
-/* rizin - LGPL - Copyright 2017-2018 - wargio, pancake */
+// SPDX-FileCopyrightText: 2017-2018 wargio <wargio@libero.it>
```
```suggestion
// SPDX-FileCopyrightText: 2017-2018 deroad <wargio@libero.it>
```

XVilka commented on librz/asm/arch/rsp/rsp_idec.c:
```diff
@@ -1,3 +1,4 @@
+// SPDX-FileCopyrightText: 2016 bobby.smiles32@gmail.com <bobby.smiles32@gmail.com>
```
Same here, remove `@gmail.com` from the name

wargio commented on librz/util/pkcs7.c:
```diff
@@ -1,4 +1,5 @@
-/* rizin - LGPL - Copyright 2017-2018 - wargio */
+// SPDX-FileCopyrightText: 2017-2018 wargio <wargio@libero.it>
```
```suggestion
// SPDX-FileCopyrightText: 2017-2018 deroad <wargio@libero.it>
```

wargio commented on librz/util/protobuf.c:
```diff
@@ -1,4 +1,5 @@
-/* rizin - LGPL - Copyright 2017-2019 - wargio */
+// SPDX-FileCopyrightText: 2017-2019 wargio <wargio@libero.it>
```
```suggestion
// SPDX-FileCopyrightText: 2017-2019 deroad <wargio@libero.it>
```

wargio commented on librz/util/x509.c:
```diff
@@ -1,4 +1,5 @@
-/* rizin - LGPL - Copyright 2017-2018 - wargio */
+// SPDX-FileCopyrightText: 2017-2018 wargio <wargio@libero.it>
```
```suggestion
// SPDX-FileCopyrightText: 2017-2018 deroad <wargio@libero.it>
```

ret2libc committed:
SPDX Copyright text for all files based on history

- Add LICENSES directory
- Download additional licenses used
- Add .reuse directory
- Add doc about SPDX/reuse

Co-authored-by: Florian Märkl <info@florianmaerkl.de>

wargio commented on librz/core/cmd_descs/cmd_descs_generate.py:
```diff
@@ -1,12 +1,15 @@
 import yaml
 
-CMDDESCS_C_TEMPLATE = """// SPDX-License-Identifier: LGPL-3.0-only
+CMDDESCS_C_TEMPLATE = """// SPDX-FileCopyrightText: 2020 RizinOrg <info@rizin.re>
```
`SPDX-FileCopyrightText: 2020-{current_year} RizinOrg <info@rizin.re>` mabye

ret2libc replied:
I just didn't want to update this every year :D I don't think it's mandatory to have a range

wargio replied:
i think this is wrong. we should update this based on changes. so every time we update the commands, the files needs to be regenerated (therefore also the header like you normally would do for any fix).

ret2libc replied:
I put RIzinOrg instead of my name or the name of someone else to avoid having to adapt this line every time one adds a bunch of commands.

XVilka commented:
Could you please add a paragraph about those licenses/SPDX headers in:
- PACKAGERS.md
- CONTRIBUTORS.md (or DEVELOPERS.md)

XVilka commented on librz/asm/arch/include/opcode/mips.h:
```diff
@@ -1,3 +1,6 @@
+// SPDX-FileCopyrightText: 1993, 1994, 1995, 1996, 1997, 1998, 1999, 2000, 2001, 2002, 2003, 2004, 2005, 2008 Free Software Foundation, Inc.
+// SPDX-License-Identifier: GPL-1.0-or-later
```
Looks like this file is very old and should be updated. Latest GDB uses GPLv3-based one...
See https://github.com/rizinorg/rizin/issues/209

ret2libc replied:
I know... of course for another PR. THis one is becoming tiring already this way :(

XVilka replied:
Yes, I wasn't implying to do that, just never noticed before and worth noting.

XVilka commented on librz/asm/arch/include/opcode/tricore.h:
```diff
@@ -1,3 +1,6 @@
+// SPDX-FileCopyrightText: 1998-2003 Free Software Foundation, Inc.
+// SPDX-License-Identifier: GPL-1.0-or-later
```
Same here, very old one.

thestr4ng3r commented on .reuse/dep5:
```diff
@@ -0,0 +1,198 @@
+
+Files: doc/img/rizin.svg
+Copyright: 2021 RizinOrg <info@rizin.re>
+License: CC0-1.0
```
This should be
```suggestion
License: LGPL-3.0-only OR CC-BY-SA-4.0
```

thestr4ng3r reviewed (changes requested):
I would not use Public Domain on anything. Better put `LGPL-3.0-only` if in doubt.

ret2libc commented:
> I would not use Public Domain on anything. Better put `LGPL-3.0-only` if in doubt.

Ok, done. Only CC0 remaining is for librz/search/aes-find.c, because it is not our code and the author said `This source is public domain. Feel free to use it and distribute it`
